### PR TITLE
Implemented getter to expose current url strategy for web plugins

### PR DESCRIPTION
**Pull request by TheOneWithTheBraid**
Implemented a getter exposing the currently present UrlStrategy for the web platform as part of `flutter_web_plugins`.
This is e.g. required to fix the below-stated issue in the `url_launcher` which is not able to properly encode an in-app URL inside an HTML `anchor` element's `href`.

*List which issues are fixed by this PR. You must list at least one issue.*
- https://github.com/flutter/flutter/issues/87941

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt.
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter_web_plugins/lib/src/navigation/url_strategy.dart [hunk 1]
```diff
@@ -9,10 +9,25 @@ import 'dart:ui' as ui;
 import 'js_url_strategy.dart';
 import 'utils.dart';
 
+/// Saves the current [UrlStrategy] to be accessed by [urlStrategy] or
+/// [setUrlStrategy].
+///
+/// This is particularly required for web plugins relying on valid URL
+/// encoding.
+UrlStrategy? _urlStrategy;
```
Comment (mdebbar) [inline packages/flutter_web_plugins/lib/src/navigation/url_strategy.dart:17]: This remains null if the app doesn't call `setUrlStrategy`.

**Comment by TheOneWithTheBraid** [comment]
Wouldn't it be nicer anyway if the `EngineFlutterWindow` directly called `setUrlStrategy` instead of the hard-coded `UrlStrategy` in `_createDefaultUrlStrategy` in https://github.com/flutter/engine/blob/2db431944795e9c556881d77cfefe7159bb58168/lib/web_ui/lib/src/engine/window.dart#L358?

**Comment by TheOneWithTheBraid** [comment]
I am unsure whether / how I can cover the affiliation with https://github.com/flutter/engine/blob/master/lib/web_ui/lib/src/engine/window.dart#L360 in a test. Do you have an idea of it?

**Review by mdebbar — APPROVED** [review]
LGTM

**Comment by mdebbar** [comment]
I'm not sure why some of the checks are stuck. You may need to rebase and push to get them to run again.

**Review by mdebbar — CHANGES_REQUESTED** [review]
PR needs clean up before submitting.

**Review by mdebbar — APPROVED** [review]
Everything is green and ready to go!! 🎉

**Pull request merged**